src/modules/aggsig/tests_impl.h

```diff
@@ -0,0 +1,192 @@
+/**********************************************************************
+ * Copyright (c) 2017 Andrew Poelstra                                 *
+ * Distributed under the MIT software license, see the accompanying   *
+ * file COPYING or http://www.opensource.org/licenses/mit-license.php.*
+ **********************************************************************/
+
+#ifndef _SECP256K1_MODULE_AGGSIG_TESTS_
+#define _SECP256K1_MODULE_AGGSIG_TESTS_
+
+#include "secp256k1_aggsig.h"
+
+void test_aggsig_api(void) {
+    /* Setup contexts that just count errors */
+    secp256k1_context *none = secp256k1_context_create(SECP256K1_CONTEXT_NONE);
+    secp256k1_context *sign = secp256k1_context_create(SECP256K1_CONTEXT_SIGN);
+    secp256k1_context *vrfy = secp256k1_context_create(SECP256K1_CONTEXT_VERIFY);
+    secp256k1_context *both = secp256k1_context_create(SECP256K1_CONTEXT_SIGN | SECP256K1_CONTEXT_VERIFY);
+    unsigned char seckeys[5][32];
+    secp256k1_pubkey pubkeys[5];
+    secp256k1_aggsig_partial_signature partials[5];
+    secp256k1_aggsig_context *aggctx;
+    unsigned char seed[32];
+    unsigned char sig[64];
+    unsigned char *msg = seed;  /* shh ;) */
+    int32_t ecount = 0;
+    size_t i;
+
+    secp256k1_context_set_error_callback(none, counting_illegal_callback_fn, &ecount);
+    secp256k1_context_set_error_callback(sign, counting_illegal_callback_fn, &ecount);
+    secp256k1_context_set_error_callback(vrfy, counting_illegal_callback_fn, &ecount);
+    secp256k1_context_set_error_callback(both, counting_illegal_callback_fn, &ecount);
+    secp256k1_context_set_illegal_callback(none, counting_illegal_callback_fn, &ecount);
+    secp256k1_context_set_illegal_callback(sign, counting_illegal_callback_fn, &ecount);
+    secp256k1_context_set_illegal_callback(vrfy, counting_illegal_callback_fn, &ecount);
+    secp256k1_context_set_illegal_callback(both, counting_illegal_callback_fn, &ecount);
+
+    for (i = 0; i < 5; i++) {
+        secp256k1_scalar tmp_s;
+        random_scalar_order_test(&tmp_s);
+        secp256k1_scalar_get_b32(seckeys[i], &tmp_s);
+        CHECK(secp256k1_ec_pubkey_create(ctx, &pubkeys[i], seckeys[i]) == 1);
+    }
+
+    aggctx = secp256k1_aggsig_context_create(none, pubkeys, 10, seed);
```

Comment:
This line makes address sanitizer complain loudly because it is an out of bounds read (pubkeys only has 5 elements).